packages/gatsby-theme-newrelic/src/hooks/useInstrumentedHandler.js

```diff
@@ -0,0 +1,35 @@
+import { useCallback, useEffect, useRef } from 'react';
+import warning from 'warning';
+
+const useInstrumentedHandler = (handler, attributes) => {
+  const savedHandler = useRef();
+
+  useEffect(() => {
+    savedHandler.current = handler;
+  }, [handler]);
+
+  return useCallback(
+    (...args) => {
+      const { actionName, ...attrs } =
+        typeof attributes === 'function' ? attributes(...args) : attributes;
+
+      if (window.newrelic) {
```

Comment:
Do we need to double check that we have access to `window` here?

Reply:
We shouldn't need to only because this function will only ever be called as part of a handler for a component, and will not be executed until the function is called.

Unless we come across a really unique use case, 99.99% of the uses of this hook will be something like the following:

```js
const handleClick = useInstrumentedHandler(() => console.log('clicked'), { actionName: 'click' })

<div onClick={handleClick} />
```

That `handleClick` doesn't get executed on a server render, so we should be safe.